mindsdb_native/libs/phases/data_splitter/data_splitter.py

```diff
@@ -83,10 +83,14 @@ def run(self, test_train_ratio=CONFIG.TEST_TRAIN_RATIO):
                 historical_test = deepcopy(self.transaction.input_data.test_df)
                 historical_test['make_predictions'] = [False] * len(historical_test)
 
-                self.transaction.input_data.test_df['make_predictions'] = [True] * len(self.transaction.input_data.test_df)
+                if self.transaction.lmd['tss'].get('nr_predictions', None):
+                    n_preds_cutoff = self.transaction.lmd['tss']['nr_predictions'] - 1
+                else:
+                    n_preds_cutoff = 0
+                self.transaction.input_data.test_df['make_predictions'] = [True] * (len(self.transaction.input_data.test_df) - n_preds_cutoff) + [False] * n_preds_cutoff
                 self.transaction.input_data.test_df = pd.concat([self.transaction.input_data.test_df,historical_train])
 
-                self.transaction.input_data.validation_df['make_predictions'] = [True] * len(self.transaction.input_data.validation_df)
+                self.transaction.input_data.validation_df['make_predictions'] = [True] * (len(self.transaction.input_data.validation_df) - n_preds_cutoff) + [False] * n_preds_cutoff
```

Comment:
I'm not 100% sure what's happening here. As in, why would we want to *not* make predictions for a certain amount of rows if we are predicting `n` points ahead ?

Reply:
See #248. However, it's not relevant anymore, the changes in time series creation avoid the new label introduction entirely, so the issue doesn't happen in the first place :)